### PR TITLE
make real_registered_proxy_url take the correct port

### PR DESCRIPTION
**Pull request by cristifalcas**

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -411,6 +411,12 @@
   }
   validate_absolute_path($puppet_cache_location)
 
+  if $registered_proxy_url {
```
Comment (ekohl) [inline manifests/init.pp]: You can shorten this to ```$real_registered_proxy_url =pick($registered_proxy_url, "https://${::fqdn}:${ssl_port}")```

**Comment by ekohl** [comment]
I'm not a great fan of this pattern, but puppet has no solution for this problem. Not sure we should fix this though since there are a lot of dependent parameters. @domcleal @mmoll any preference?

**Comment by mmoll** [comment]
I'm also really unsure. Can this somehow get catched and a warning thrown in such a case?

**Comment by ehelms** [comment]
@cristifalcas can you revisit this? We are finding we need this change as part of some Katello on Foreman work to extract the wrapped puppet foreman_proxy from our puppet-capsule. If you don't have time, please let me know and I can take over getting this change in.957748

**Comment by cristifalcas** [comment]
Hi @ehelms , I though that this PR was rejected, so I didn't push for it anymore. I'm not sure what do you mean by "can you revisit this". Just to implement the shorter notation recommended by @ekohl ?

**Comment by ekohl** [comment]
@cristifalcas I think that's the best solution, maybe add some tests as well?

**Comment by cristifalcas** [comment]
Sure, but I will not be able to do it today. Sometime next week, if it's ok with you.

**Comment by ehelms** [comment]
:+1: that would be great, thanks

**Comment by ehelms** [comment]
@cristifalcas checking back -- any chance you could tackle this in the next day or two? Otherwise, I'd like to open a PR myself with the changes.

**Comment by cristifalcas** [comment]
I don't know exactly how, but I managed to ruin this branch. I will do a new PR for this
